src/drivers/cmakeLegacyDriver.ts

```diff
@@ -103,15 +114,15 @@ export class CMakeLegacyDriver extends CMakeDriver {
         } else {
             log.debug(localize('invoking.cmake.with.arguments', 'Invoking CMake {0} with arguments {1}', cmake, JSON.stringify(args)));
             const res = await this.executeCommand(cmake, args, outputConsumer, {
-                environment: await this.getConfigureEnvironment(),
-                cwd: binary_dir
+                environment: await this.getConfigureEnvironment(configurePreset),
+                cwd: binaryDir
             }).result;
             log.trace(res.stderr);
             log.trace(res.stdout);
-            if (res.retc === 0) {
+            if (res.retc === 0 && !configurePreset) {
                 this._needsReconfigure = false;
+                await this._reloadPostConfigure();
```

Comment:
reloadPostConfigure didn't used to be dependent on the return code.  I think it would be better to put it back, but guard with a `if (!configurePreset)`

```
if (res.retc === 0 && !configurePreset) {
  this._needsReconfigure = false;
}
if (!configurePreset) {
  await this._reloadPostConfigure();
}

Reply:
that's right